newlib/libc/tinystdio/stdio.h

```diff
@@ -82,6 +82,9 @@ typedef __uint16_t __ungetc_t;
 struct __file {
 	__ungetc_t unget;	/* ungetc() buffer */
 	__uint8_t  flags;	/* flags, see below */
+#ifdef _PRINTF_PERCENT_N
```

Comment:
This adds a field to the global FILE struct when it is only needed temporarily inside the printf code?

Reply:
Hrm. glibc doesn't truncate %n values in snprintf? Reading the C spec doesn't make this entirely clear, although your interpretation is the most natural to my reading of this text in the fprintf specification:

> The number of characters written to the output stream so far by this call to fprintf

combined with this text in the snprintf specification:

> Otherwise, output characters beyond the n-1st are discarded rather than being written to the array

I'm strongly tempted to leave the existing glibc-compatible interpretation in place, even if it doesn't seem entirely aligned with the C standard as I can see where a reasonable person would say that the 'discarded' characters are part of the snprintf output.

Reply:
So, do you preferer to revert this change and leave the existing glibc-compatible interpretation in place?

Reply:
yeah, I don't think this particular rigid adherence to a slightly vague corner case in the spec is warranted, especially given that I haven't found another C library which implements it this way. That's not to say that I think you're reading is wrong; I agree that your reading is "more correct", just that it's hard to argue for a significantly more complicated implementation which isn't compatible with any other implementation...